contracts/protocol/v1/oracles/P1OracleInverter.sol

```diff
@@ -72,13 +82,27 @@ contract P1OracleInverter is
         view
         returns (uint256)
     {
+        bytes32 oracleAndAdjustment = _MAPPING_[msg.sender];
         require(
-            msg.sender == _READER_,
+            oracleAndAdjustment != bytes32(0),
             "P1OracleInverter: Sender not authorized to get price"
         );
-        uint256 rawPrice = I_P1Oracle(_ORACLE_).getPrice();
+        (address oracle, uint256 adjustment) = splitOracleAndAdjustment(oracleAndAdjustment);
+        uint256 rawPrice = I_P1Oracle(oracle).getPrice();
         uint256 invertedPrice = rawPrice.baseReciprocal();
-        uint256 result = invertedPrice.baseMul(_ADJUSTMENT_);
+        uint256 result = invertedPrice.baseMul(adjustment);
         return result;
     }
+
+    function splitOracleAndAdjustment(
+        bytes32 oracleAndAdjustment
+    )
+        private
+        pure
+        returns (address, uint256)
+    {
+        address oracle = address(bytes20(oracleAndAdjustment));
+        uint256 adjustment = uint256(oracleAndAdjustment & ADJUSTMENT_MASK);
```

Comment:
nit: rather than `ADJUSTMENT_MASK`, perhaps `uint256(uint96(uint256(oracleAndAdjustment)))` is clearer?